packages/clay-layout/src/__tests__/__snapshots__/index.tsx.snap

```diff
@@ -0,0 +1,146 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`ClayLayout renders 1`] = `
+<div>
+  <div
+    class="container"
```

Comment:
As I said above, if we want to somewhat match expectations, we might want to be a tad bit opinionated here and have this be `container-fluid container-fluid-xl` which is by far our most beloved container 😂

Reply:
I think I would push back on that for the library level. Unless you can think of a good naming convention for the component to indicate it will include those. For a user to use `<Container>` without props and then get those classes seems odd to me. On the portal side of things, we can default to that since it makes sense for that to be portal's default, but not sure it makes sense to default at the library level.

I would also potentially push back and add that if we expect the default behavior of container to be `container-fluid container-fluid-xl` we might consider making that at the css level.

Reply:
I agree with @bryceosterhaus here. Default values can be provided in the implementation on Portal, thus preserving abstraction for the end-user while still providing users of Clay as a library the flexibility.

Reply:
Hey @bryceosterhaus, @kresimir-coko, I'll try to go at some more length in the general comments, but:

Once again, I think we're failing to see that right now, 100% (unless Faro moved to Clay) of our users are DXP. Also, **90% of our usage when implementing lexicon is `container-fluid`.** This means that by not providing this we're refusing to provide **real value to real users** of our library based on some not-so-real user/library expectation. Since 90% of the times someone in Liferay looking to build a UI had to use `container-fluid`, it is our goal to make this easier for them.

Please, go through:
- https://github.com/jbalsas/liferay-portal/pull/2160/commits/e72f30c231e196b72dcf15bbff7d94ddda6b2655
- https://github.com/eduardoallegrini/liferay-portal/pull/71/commits
- https://github.com/eduardoallegrini/liferay-portal/pull/72/commits/4c9f07b95116fd6f27dba4f59e6e0557299bd29d

The main goal of Clay is to **simplify implementing Lexicon UIs**. The main implementor of Lexicon UIs is DXP. Based on the above changes, I'd say those are better defaults for implementing Lexicon UIs. Thus, those defaults belong in Clay, not in DXP. Otherwise, we're not making it any easier. There are literally thousands of occurrences of `container-fluid` and around tens of `container`. It doesn't make any sense that we'd favour the latter here.

If you want to be real purist here, **let's maybe create `FluidContainer` that composes `Container` and provides the right semantics and ergonomics to our devs**. We can do the same in the Portal side and mostly simply use `<clay:fluid-container>` and `<ClayLayout.FluidContainer>`.

Reply:
@wincent, @marcoscv-work, thoughts?

Reply:
@jbalsas When you put it like that yea, I agree. I guess I didn't understand the exact goal of Clay, I thought it was a low-level library first, DXP-specific second. 

I figured people using DXP used Clay to build their apps, but I assume it's not like that after all. I don't personally care about being a purist here, so I have no problem with having the default be `container-fluid container-fluid-xl`. 

Something to maybe think about is that we can have a prop that strips all custom classes from the container, like `<ClayContainer basic/vanilla/lite />`, this will provide the defaults for 90% of the users in DXP like you said, but also give the dev that's using Clay for a personal project a single prop to enable flexibility in styling via CSS with just a single prop.

Reply:
> I didn't understand the exact goal of Clay, I thought it was a low-level library first, DXP-specific second

<img width="799" alt="Screen Shot 2020-05-11 at 09 44 02" src="https://user-images.githubusercontent.com/905006/81536536-06ea7080-936c-11ea-89f2-a0596fa87559.png">

That 👆 😂 

> `<ClayContainer basic/vanilla/lite />`

You were earlier arguing that you wouldn't bother checking props and go about it your way with CSS, so why the change of heart? 

I can understand @bryceosterhaus's struggle here, so I'm not pushing hard to make those the defaults for `<ClayContainer>`.

We need to:
- Fulfil the library's goal which is making Lexicon UIs easier
- We need to analyze what it means to build Lexicon UIs. Our biggest source of data here is DXP since that's the main Lexicon UI implementor. 
- We need to provide defaults and components based on how real developers are developing real UIs. That's the fastest way to provide real value to the users of our library.

Reply:
You got me there @jbalsas with that screenshot 😆 

### Why the change of heart? 
Not really a change of heart I think, if it's one prop that will disable all custom classes except `.container` I think is a good compromise. If that prop shows up on autocomplete as one of the first ones that's even better. But then again, my PoV was different when we started this discussion, as I was looking at it from a different use-case than I am now. Now I agree with you @jbalsas.

### Fluid Container
I think that having a FluidContainer on top of Container might muddy things up unnecessarily, as you said, 90% of usages are Fluid, and we can maybe add a `notFluid` prop that will omit Fluidity from the container. I would really dislike going into DXP code a couple years from now and seeing `<FluidContainer />` everywhere

Reply:
> @wincent, @marcoscv-work, thoughts?

This is one of those ones where I don't have anything new to add beyond what I've already said (here, [and on the issue](https://github.com/liferay/clay/issues/2421))  — and in other places about [what/why I think Clay even is](https://github.com/liferay/clay/pull/3034#issuecomment-603125574) — , and the discussion here is already very difficult to follow so I won't help much by adding more comments.

Reply:
> This means that by not providing this we're refusing to provide real value to real users of our library based on some not-so-real user/library expectation. 

Yeah this totally makes sense, my thought is based more css vs react component discrepancies. **My main concern is that we should have consist naming in our components that stem from @clayui/css.** By defaulting `ClayLayout.Container` to be fluid, seemed to be at odds with @clayui/css. Also, if Lexicon is recommending the styles of `.container-fluid` to be used 90% of the time, and `.container` just 10% of the time, maybe we should just have one class and not use the non-fluid styles.

> If you want to be real purist here, let's maybe create FluidContainer that composes Container and provides the right semantics and ergonomics to our devs. 

I'm not sure I would label myself as a _purist_ 😂, I just want to make sure we are consistent from @clayui/css to our react components. I like this idea for naming though. Maybe even going with `ClayLayout.ContainerFluid` for more consistency. I think this would be inline with both @clayui/css and the need in DXP.

Reply:
> @kresimir-coko 
> I would really dislike going into DXP code a couple years from now and seeing <FluidContainer /> everywhere


What would be the issue with this? I don't think it'lI be an issue really and I think it will actually be better for readability in the future and knowing exactly what the containers purpose is. I also don't mind it because it'll be easy to find/replace for any future refactors

Reply:
> @wincent, @marcoscv-work, thoughts?

We could create `FluidContainer` and `Container` to fit semantic but finally, we wouldn't use `Container` for anything, so I'm closer to fix the semantic in future versions of CSS than creating components or variations we don't really need

Reply:
Actually, when we weigh in our `autofit` rows and cols this might play a part since those do more or less need an inert container... so `Container` (without the "container"  class actually... 😂) could help bridge the gave between the different grids.

Or maybe something else, so we'd have

Layout (Stack)  -> Container/FluidContainer -> Row/Col
                          |-> Row autofit -> Col autofit

Adding one additional layer feels a bit over-engineered... so maybe we could simplify with:

```jsx
<Container> -> <div class="container">...</div>
<Container autofit> -> <div>...</div>
<FluidContainer> -> <div class="container-fluid container-fluid-max-xl">...</div>
```

Reply:
Container 'type' autofit sounds good to me, if it makes clear this is not the BS container is enough.

It is out of time but I was thinking about something more semantic which wraps everything, something like:

Using **<Grid>**
<Grid container>
<Grid row>
<Grid col>
<Grid autofit-row>
<Grid autofit-col>